src/robot/utils/connectioncache.py

```diff
@@ -137,13 +152,14 @@ def __len__(self):
     def __bool__(self):
         return self.current is not self._no_current
 
-    def resolve_alias_or_index(self, alias_or_index):
-        for resolver in self._resolve_alias, self._resolve_index:
+
+    def _resolve_connection(self, alias_or_index):
```

Comment:
This method is part of the public API (see #3125) and thus cannot be renamed. Now that we support also connections themselves, `resolve_alias_or_index` is not a totally accurate name, but renaming it would require first a deprecation period. I personally consider this name good enough and would perhaps only update documentation. If you want to change the name, let's do that as a separate PR.

Reply:
reverted method name change , we can figure that in another PR